src/routes/usergroups.ts

```diff
@@ -1,5 +1,323 @@
 import * as db from '../tools/db';
-import { Application } from "express";
+import { Application, Request, Response } from 'express';
 
 export const register = (app: Application) => {
+    /* /usergroup endpoint */
+    
+    // Get method
+    app.get('/api/v1/usergroup', (req: Request, res: Response) => {
+        db.connection.query('SELECT * FROM UserGroup', (error, results, fields) => {
+            if (error) {
+                return res.sendStatus(500);
+            }   
+            return res.send(results);
+        });
+    });
+
+    // Post method
+    app.post('/api/v1/usergroup', (req: Request, res: Response) => {
+        let title = req.query.title;
+        let descrip = req.query.descrip;
```

Comment:
I just saw all of these are query parameters but especially for POST, PUT this is a bad practice. In this case, you should get `title`, `descrip` and `max_capacity` from the JSON POST content instead